### PR TITLE
Remove unused function

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
bors r-

Used in tests, that's why it wasn't caught by x.py check

**Comment by matklad** [comment]
Moved the tests, ready for review

File: src/librustc_lexer/src/tests.rs [hunk 1]
```diff
@@ -128,6 +128,11 @@ fn check_lexing(src: &str, expect: Expect) {
     expect.assert_eq(&actual)
 }
 
+#[test]
+fn smoke_test() {
+    check_lexing("/* my source file */ fn main() { println!(\"zebra\"); }\n", expect![[]])
```
Comment (petrochenkov) [inline src/librustc_lexer/src/tests.rs:133]: >expect![[]]

The test output wasn't updated?

**Comment by petrochenkov** [comment]
r=me with https://github.com/rust-lang/rust/pull/76050#discussion_r479636065 addressed

File: src/librustc_lexer/src/tests.rs (hunk 1 above)
Comment (matklad) [reply src/librustc_lexer/src/tests.rs]: Ah indeed... At least now we now that these tests are actually capable of failing on CI:

https://github.com/rust-lang/rust/runs/1044897833?check_suite_focus=true#step:23:2341

**Comment by matklad** [comment]
Rebased over the great compiler renaming

@bors r=petrochenkov

**Pull request merged**